vega_sim/scenario/common/agents.py

```diff
@@ -1473,6 +1475,301 @@ def _calculate_price_volume_levels(
+        self.external_delay = external_delay
+
+        self.transfer_threshold = transfer_threshold
+
+    def initialise(
+        self,
+        vega: Union[VegaServiceNull, VegaServiceNetwork],
+        create_wallet: bool = True,
+        mint_wallet: bool = True,
+    ):
+        # Initialise the parent ExponentialShapedMarketMaker
+        super().initialise(vega, create_wallet, mint_wallet)
+
+        self.external_market_id = self.vega.find_market_id(
+            name=self.external_market_name
+        )
+        self._update_state(current_step=self.current_step)
+
+        if vega.create_wallet:
+            vega.create_wallet(
+                name=self.wallet_name,
+                passphrase=self.wallet_pass,
+                key_name=self.external_key_name,
+            )
+        if mint_wallet:
+            vega.mint(
+                wallet_name=self.wallet_name,
+                asset=self.asset_id,
+                amount=self.initial_asset_mint,
+                key_name=self.external_key_name,
+            )
+
+    def _update_state(self, current_step: int):
+        super()._update_state(current_step)
+
+        if self.state_update_freq and current_step % self.state_update_freq == 0:
+
+            int_market_info = (
+                self.vega.market_info(market_id=self.market_id)
+                if self.market_id is not None
+                else None
+            )
+            self.int_mkr_fee = (
+                float(int_market_info.fees.factors.maker_fee)
+                if int_market_info is not None
+                else 0
+            )
+            self.int_liq_fee = (
+                float(int_market_info.fees.factors.liquidity_fee)
+                if int_market_info is not None
+                else 0
+            )
+            self.int_inf_fee = (
+                float(int_market_info.fees.factors.infrastructure_fee)
+                if int_market_info is not None
+                else 0
+            )
+
+            ext_market_info = (
+                self.vega.market_info(market_id=self.external_market_id)
+                if self.external_market_id is not None
+                else None
+            )
+            self.ext_mkr_fee = (
+                float(ext_market_info.fees.factors.maker_fee)
+                if ext_market_info is not None
+                else 0
+            )
+            self.ext_liq_fee = (
+                float(ext_market_info.fees.factors.liquidity_fee)
+                if ext_market_info is not None
+                else 0
+            )
+            self.ext_inf_fee = (
+                float(ext_market_info.fees.factors.infrastructure_fee)
+                if ext_market_info is not None
+                else 0
+            )
+
+    def _optimal_strategy(self, current_position, current_step):
+
+        ext_best_bid, ext_best_ask = self.vega.best_prices(
+            market_id=self.external_market_id
+        )
+
+        int_fee = self.int_mkr_fee + self.int_liq_fee
```

Comment:
I think this is only going to be true in the case where the MM is the *only* MM so that all liquidity fees go to them. Worth being aware of if so

Reply:
Ah that is a very good point that could be tricky, not sure what gRPC request we could use to estimate ELS either..?

Reply:
After an epoch you can pull the calculated one, but obviously most of our runs don't really go that long! So we could either shorten epochs or just leave it for now 😄. For the purpose of the sim we're actually using this for it's correct anyway. Maybe just spin up an issue for it so we remember. (One half way house might be to just use the ratio of liquidity provision over all liquidity provisions, assuming they'll all continue to exist)

Reply:
Sounds good thanks, have raised this issue [here](https://github.com/vegaprotocol/vega-market-sim/issues/286) 👍